manim/mobject/number_line.py

```diff
@@ -89,6 +88,11 @@ def __init__(
             self.unit_size = self.get_unit_size()
         self.shift(-self.number_to_point(self.number_at_center))
 
+        self.numbers_with_elongated_ticks = []
+        for number in numbers_with_elongated_ticks:
+            if number >= self.x_min and number <= self.x_max:
+                self.numbers_with_elongated_ticks.append(number)
```

Comment:
This could be condensed into a single list comprehension, but it's just a suggestion and I won't block the PR over it.

```suggestion
        self.numbers_with_elongated_ticks = [
            nbr for nbr in numbers_with_elongated_ticks if self.x_min <= nbr <= self.x_max
        ]
```

Reply:
I retested the code with this change, both pytest and visual inspection.  All good, so I accepted the suggested code change.